gridded_etl_tools/utils/metadata.py

```diff
@@ -767,15 +793,21 @@ def remove_unwanted_fields(self, dataset: xr.Dataset) -> xr.Dataset:
         dataset : xarray.Dataset
             The dataset being published, pre-metadata update
         """
-        compressor = numcodecs.Blosc() if self.use_compression else None
-
         for coord in ["latitude", "longitude"]:
             dataset[coord].attrs.pop("chunks", None)
             dataset[coord].attrs.pop("preferred_chunks", None)
             dataset[coord].encoding.pop("_FillValue", None)
             dataset[coord].encoding.pop("missing_value", None)
-            dataset[coord].encoding["compressor"] = compressor
         dataset[self.data_var].encoding.pop("filters", None)
+
+    def apply_compression(self, dataset: xr.Dataset):
```

Comment:
I think we want to call this something other than "apply". If I'm understanding, even if this does end up applying compression, it doesn't apply it immediately. And if there already is compression, setting this doesn't do anything, right?

Maybe "set compression encoding" or something like that would make sense. I think this docstring would be a good place to put details about what actually happens, like it will be applied when writing the Zarr if there is no compression already. And if you need to modify compression, use the modify function.

I'm wondering whether it would make sense to raise an error if this is used on a dataset with compression already set. Wouldn't you always need to use modify in that case?